### ISSUE TITLE
Option for using up and down arrow keys for text navigation rather than history navigation

### ISSUE THREAD

**Issue by briankendall** [thelounge/thelounge#4443]
### Feature Description

When typing, I generally like to use the up and down arrow keys for moving the text cursor up and down a line in the text field, or if there's just a single line, navigating to the beginning or end of the field (as is the convention in macOS). But The Lounge uses up and down for navigating through your message history. I understand that's something of a tradition in IRC clients, harkening back to the days where everything was in a terminal window.

How about an option that allows switching between these two behaviors, defaulting to the current behavior? If I hacked that feature in and made a PR, would it be welcome?

**Comment by dgw** [comment]
#4205?

**Comment by briankendall** [comment]
That's odd, it's not working for me. Pressing up always go up in history, regardless of how many lines there are in the text field. So far I've seen that behavior in Ungoogled Chromium 92.0.4515.131 and Ferdi on macOS.

Maybe then this isn't so much a feature request as it is a bug report?

**Comment by brunnre8** [comment]
The problem is that those are "virtual" lines, meaning that the input field doesn't actually contain new lines but only gets soft wrapped.
So those aren't actually lines and hence #4205 (or well, the commit that survived) doesn't solve the problem.

But even if we fix that (we should) that wouldn't solve your issue as being on the first line (which is the normal case) would still navigate the history.
That's pretty much by design and as TL tries to minimize the options available to avoid becoming to complicated I don't think that's something TL should make configurable.

There's other, browser native shortcuts that do input field navigation... Ctrl-Home in the case of beginning of field. That is on sane OSes :stuck_out_tongue_winking_eye: don't know if that translates directly to apple.

**Comment by briankendall** [comment]
Yes, macOS has some of the same shortcuts. In fact I can use home and end to move to the beginning and end of the text field, so there are already alternatives. The trouble is that the keystrokes I use for text navigation are pretty much hard wired into my brain at this point. It'd take a lot of effort to change them!

I can probably get the behavior I want by just injecting a little extra javascript into the page via an extension.
